great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -52,69 +54,63 @@ class SQLDatasourceWarning(UserWarning):
     pass
 
 
-@pydantic_dc.dataclass(frozen=True)
-class ColumnSplitter:
+class _ColumnSplitter(ExperimentalBaseModel):
     column_name: str
     method_name: str
 
+    @property
+    def param_names(self) -> List[str]:
```

Comment:
@billdirks What would be a helpful docstring explaining this interface methods for developers needing to extend it?  Thanks.

Reply:
Done.

Note, we decided in an early discussion that column splitters are not a point of extension for contributors since this is tightly couple to non-public api code, namely, [this file](https://github.com/great-expectations/great_expectations/blob/f0e89bdbaa491f39ba06f3d7af3d0b4d31d425bf/great_expectations/execution_engine/split_and_sample/data_splitter.py) and the implementations in that directory. So the typical developer extending this is the core developers.